lib/src/analysis_options.dart

```diff
@@ -25,74 +32,85 @@ class AnalysisOptions {
       @required this.rules});
 
   factory AnalysisOptions.from(String content) {
+    try {
+      final node = loadYamlNode(content ?? '');
+
+      return AnalysisOptions.fromMap(
+          node is YamlMap ? yamlMapToDartMap(node) : {});
+    } on YamlException catch (e) {
+      throw FormatException(e.message, e.span);
+    }
+  }
+
+  factory AnalysisOptions.fromMap(Map<String, Object> configMap) {
     Config metricsConfig;
-    var metricsExcludePatterns = <String>[];
+    Iterable<String> metricsExcludePatterns = <String>[];
     var rules = <String, Map<String, Object>>{};
 
-    final node = loadYamlNode(content ?? '');
-    if (node is YamlMap && node.nodes[_rootKey] is YamlMap) {
-      final metricsOptions = node.nodes[_rootKey] as YamlMap;
-
-      if (_isYamlMapOfStringsAndIntegers(metricsOptions.nodes[_metricsKey])) {
-        final configMap = metricsOptions[_metricsKey] as YamlMap;
+    final metricsOptions = configMap[_rootKey];
+    if (metricsOptions != null && metricsOptions is Map<String, Object>) {
+      final configMap = metricsOptions[_metricsKey];
+      if (configMap != null && configMap is Map<String, Object>) {
         metricsConfig = Config(
           cyclomaticComplexityWarningLevel:
-              configMap['cyclomatic-complexity'] as int,
-          linesOfCodeWarningLevel: configMap['lines-of-code'] as int,
+              configMap['cyclomatic-complexity'].as<int>(),
+          linesOfCodeWarningLevel: configMap['lines-of-code'].as<int>(),
           numberOfArgumentsWarningLevel:
-              configMap['number-of-arguments'] as int,
-          numberOfMethodsWarningLevel: configMap['number-of-methods'] as int,
+              configMap['number-of-arguments'].as<int>(),
+          numberOfMethodsWarningLevel: configMap['number-of-methods'].as<int>(),
         );
       }
 
-      if (_isYamlListOfStrings(metricsOptions.nodes[_metricsExcludeKey])) {
-        metricsExcludePatterns =
-            List.unmodifiable(metricsOptions[_metricsExcludeKey] as Iterable);
+      final excludeList = metricsOptions[_metricsExcludeKey];
+      if (excludeList != null &&
+          excludeList is Iterable<Object> &&
```

Comment:
Same suggestion as for map, but for iterable.

Reply:
fixed